user_management/api/views.py

```diff
@@ -151,7 +152,8 @@ def initial(self, request, *args, **kwargs):
         try:
             self.user = User.objects.get(pk=uid)
         except User.DoesNotExist:
-            raise Http404()
+            msg = _('Invalid or expired token.')
+            raise PermissionDenied(detail=msg)
```

Comment:
`419` fits one of the possible causes (expired token) but not the other (invalid token).

`410` doesn't feel right - what is the "resource" in this case? The token isn't really what the user is interested in - instead it's an implementation detail for controlling access to a resource that can be requested again.